lib/resources/forms.js

```diff
@@ -271,8 +271,9 @@ module.exports = (service, endpoint) => {
   ////////////////////////////////////////
   // RESTORE / UNDELETE
 
+  // Instead of the xmlFormId, this endpoint uses the numeric form id in the database
   service.post('/projects/:projectId/forms/:id/restore', endpoint(({ Forms }, { auth, params }) =>
-    Forms.getByProjectAndXmlFormId(params.projectId, params.id, false, null, QueryOptions.none, true)
+    Forms.getByProjectAndNumericId(params.projectId, params.id, false, null, QueryOptions.none, true)
```

Comment:
Broader suggestion: make all form resources able to take either numeric form id or xml form id

Maybe for now: just return form int id with deleted forms so it can be used here